ambry-commons/src/main/java/com.github.ambry.commons/ReadableStreamChannelInputStream.java

```diff
@@ -55,7 +55,7 @@ public int read()
     int data = -1;
```

Comment:
`available()` should not return -1 (which I think it can in this implementation). Also, whatever `available()` returns should be readable without blocking - so returning all of `bytesAvailable` (or Integer.MAX) would not be the right thing as `loadData()` can block in trying to read that much.